pkg/cmd/pulumi/convert/convert.go

```diff
@@ -545,10 +558,15 @@ func generateAndLinkSdksForPackages(
 			continue
 		}
 
+		packageSource := pkg.Name
+		if pkg.Version != nil {
+			packageSource += "@" + pkg.Version.String()
+		}
+
```

Comment:
Use the right package source here because using `pkg.Name` will resolve to latest even if a version is specified